coralproject/talk
# [CORL-2835] Reject all comments in queue

## What does this PR do?
This PR implements a feature in which selecting to reject all comments from a user while banning them will cause those comments to disappear from the moderation queue.

## These changes will impact:


## What changes to the GraphQL/Database Schema does this PR introduce?
The `User` type has been extended on the frontend to include a boolean flag for `allCommentsRejected`

## Does this PR introduce any new environment variables or feature flags?
No.

## If any indexes were added, were they added to `INDEXES.md`?
n/a

## How do I test this PR?
1. As a user with moderation privileges, navigate to the moderation queue.
2. Use the "b" key to open the ban modal for one of the comments' authors
3. Check the box for "reject all comments by this user"
4. Ban the user
5. Observe that any other comments by that user currently in the moderation queue should disappear.

## Where any tests migrated to React Testing Library?
No.

## How do we deploy this PR?
No special considerations should be required.
